cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ProbeGenerator.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import io.fabric8.kubernetes.api.model.ProbeBuilder;
+import io.strimzi.api.kafka.model.Probe;
+import io.strimzi.api.kafka.model.TlsSidecar;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ProbeGenerator {
+
+    private final Probe probeConfig;
+
+    private ProbeGenerator(Probe probeConfig) {
+        this.probeConfig = probeConfig;
+    }
+
+    public static ProbeGenerator of(Probe probeConfig) {
+        if (probeConfig == null) {
+            throw new IllegalArgumentException();
+        }
+        return new ProbeGenerator(probeConfig);
```

Comment:
This is really what my concern with regards to the memory was. If you are sure it has no impact I'm fine. I admit that  I have no clue how things like memory work in Java or any other language :-o.